File: 3241-divide-array-into-arrays-with-max-difference/3241-divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        sort(begin(nums), end(nums));

        int n = nums.size();
        vector<vector<int>> ans;
        for(int i = 0; i<n-2; i+=3){
            if(nums[i+2]-nums[i] > k){
                ans.clear();
                return ans;
            }
            vector<int> v;
            v.push_back(nums[i]);
            v.push_back(nums[i+1]);
            v.push_back(nums[i+2]);
            ans.push_back(v);
        }
        return ans;
    }
};

File: 3241-divide-array-into-arrays-with-max-difference/3241-divide-array-into-arrays-with-max-difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int n = nums.length;
        Arrays.sort(nums);
        int[][] ans = new int[n/3][3];
        int j = 0;
        for(int i = 0; i<n-2; i+=3){
            if(nums[i+2]-nums[i] > k){
                return new int[0][0];
            }
            ans[j++] = new int[]{nums[i], nums[i+1], nums[i+2]};
        }
        return ans;
    }
}